### ISSUE TITLE
btn-primary and btn-outline-primary does not respect --bs-primary color

### ISSUE THREAD

**Issue by harnishdesign** [twbs/bootstrap#38788]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

When I set `--bs-primary` variable in my custom CSS and use `btn-primary` and `btn-outline-primary` class, Basically ignoring the variable.

### Reduced test cases

```
:root {
  --bs-primary: red;
  --bs-primary-rgb: 255, 0, 0;
}
```

```
<a class="btn btn-primary">This should be red, but is still blue</div>
<a class="btn btn-outline-primary">This should be red, but is still blue</div>
```

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Firefox

### What version of Bootstrap are you using?

v5.3.0

**Comment by 0c3d7r4** [comment]
@julien-deramond why not? Wouldn't it make sense to reference colors defined as variables rather than hard-code them?

**Comment by julien-deramond** [comment]
Yes, it makes sense. Just saying that it's not possible to do right now because we introduced CSS variables in a first batch to avoid too much breaking changes. We'll try to refactor everything for the v6 to provide something like a nice color palette configuration where one will be able to configure more easily compared to what is actually provided.

**Comment by mdo** [comment]
Fixed in v6.
